server/src/main/java/io/druid/client/coordinator/CoordinatorClient.java

```diff
@@ -0,0 +1,126 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.client.coordinator;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Charsets;
+import com.google.common.base.Throwables;
+import com.google.inject.Inject;
+import com.metamx.common.ISE;
+import com.metamx.http.client.HttpClient;
+import com.metamx.http.client.Request;
+import com.metamx.http.client.response.StatusResponseHandler;
+import com.metamx.http.client.response.StatusResponseHolder;
+import io.druid.client.SegmentLoadInfo;
+import io.druid.client.selector.Server;
+import io.druid.curator.discovery.ServerDiscoverySelector;
+import io.druid.guice.annotations.Global;
+import org.jboss.netty.handler.codec.http.HttpMethod;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.joda.time.Interval;
+
+import java.net.URI;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.List;
+
+public class CoordinatorClient
+{
+  private static final StatusResponseHandler RESPONSE_HANDLER = new StatusResponseHandler(Charsets.UTF_8);
+
+  private final HttpClient client;
+  private final ObjectMapper jsonMapper;
+  private final ServerDiscoverySelector selector;
+
+  @Inject
+  public CoordinatorClient(
+      @Global HttpClient client,
+      ObjectMapper jsonMapper,
+      @Coordinator ServerDiscoverySelector selector
```

Comment:
I agree we should separate external from internal service discovery. However, is there any reason we can't have two `ServerDiscoverySelector`s, one for internal, and one for external purposes? It seems easier to work with and less error-prone than handling the announcement manually for pure discovery, which is part of the reason for why it was used in `RemoteTaskActionClient` instead of relying on announcer.